### PR TITLE
about model

### PR DESCRIPTION
**Pull request by azizbabul**
[taiga](https://tree.taiga.io/project/hmtanbir-college/us/20?kanban-status=844213)

### CODE REVIEW

File: app/views/admin/abouts/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  <div class="field">
+    <%= f.label :slug %>
+    <%= f.text_field :slug %>
+  </div>
+
+  <div class="field">
+    <%= f.label :description %>
+    <%= f.text_area :description %>
+  </div>
+
```
Comment (hmtanbir) [inline app/views/admin/abouts/_form.html.erb]: Please add field for image.

File: app/views/admin/abouts/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      <th>Description</th>
+      <th colspan="3"></th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <% @admin_abouts.each do |admin_about| %>
+      <tr>
+        <td><%= admin_about.slug %></td>
+        <td><%= admin_about.description %></td>
```
Comment (hmtanbir) [inline app/views/admin/abouts/index.html.erb]: Add image thumbs in this page.

File: app/views/admin/abouts/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+<p>
+  <strong>Slug:</strong>
+  <%= @admin_about.slug %>
+</p>
+
+<p>
+  <strong>Description:</strong>
+  <%= @admin_about.description %>
+</p>
+
```
Comment (hmtanbir) [inline app/views/admin/abouts/show.html.erb]: Add image medium in this page.

File: config/routes.rb [hunk 4]
```diff
@@ -1,5 +1,8 @@
 Rails.application.routes.draw do
   namespace :admin do
+    resources :abouts
```
Comment (hmtanbir) [inline config/routes.rb]: Use DRY.

File: spec/controllers/admin/abouts_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+  let(:valid_attributes) do{
+
+
+  slug: 'first semester',
+      description: '1st'
+  }
+end
+  let(:invalid_attributes) do{
+    slug: '',
+        description: '1st'
```
Comment (hmtanbir) [inline spec/controllers/admin/abouts_controller_spec.rb]: Use invalid data.

File: spec/models/admin/about_spec.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+require 'rails_helper'
+
+RSpec.describe Admin::About, type: :model do
+
+
+  it { is_expected.to validate_presence_of :slug }
```
Comment (hmtanbir) [inline spec/models/admin/about_spec.rb]: add uniqueness test.

File: spec/views/admin/abouts/index.html.erb_spec.rb [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+        :description => "MyText"
+      ),
+      Admin::About.create!(
+        :slug => "Slug",
+        :description => "MyText"
+      )
+    ])
+  end
+
+  it "renders a list of admin/abouts" do
```
Comment (hmtanbir) [inline spec/views/admin/abouts/index.html.erb_spec.rb]: Please fix this test.

File: app/models/admin/about.rb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+class Admin::About < ApplicationRecord
+  has_attached_file :image, styles: { large:'600x600>',medium: '300x300>', thumb: '100x100>' }, default_url: '/assets/rails.png'
+  validates_attachment_content_type :image, content_type: /\Aimage\/.*\Z/
+validates_presence_of :slug
```
Comment (hmtanbir) [inline app/models/admin/about.rb]: slug must be unique and description will be validated.

File: spec/models/admin/about_spec.rb (hunk 6 above)
Comment (azizbabul) [inline spec/models/admin/about_spec.rb]: unique ness added

File: spec/controllers/admin/abouts_controller_spec.rb (hunk 5 above)
Comment (azizbabul) [inline spec/controllers/admin/abouts_controller_spec.rb]: invalid data us already used as slug: ' ' as this cannot be empty

File: app/views/admin/abouts/index.html.erb (hunk 2 above)
Comment (azizbabul) [inline app/views/admin/abouts/index.html.erb]: added

File: app/models/admin/about.rb (hunk 8 above)
Comment (azizbabul) [inline app/models/admin/about.rb]: unique atrirbute added

File: app/views/admin/abouts/index.html.erb (hunk 2 above)
Comment (azizbabul) [inline app/views/admin/abouts/index.html.erb]: added

File: config/routes.rb (hunk 4 above)
Comment (azizbabul) [inline config/routes.rb]: DRY followed

**Review by hmtanbir — APPROVED** [review]

**Pull request merged**